### PR TITLE
Fix importing nested model files

### PR DESCRIPTION
**Pull request by rkodev**
## Overview

Fixes import statements for go imports in nested builder folders
Fixes https://github.com/microsoftgraph/msgraph-sdk-go/issues/474
### Demo

E.g  creating worklow  https://learn.microsoft.com/en-us/graph/api/identitygovernance-lifecycleworkflowscontainer-post-workflows?view=graph-rest-beta&tabs=go

Current 

```go


```
import (
	  "context"
	  msgraphsdk "github.com/microsoftgraph/msgraph-beta-sdk-go"
	  graphmodels "github.com/microsoftgraph/msgraph-beta-sdk-go/models//identityGovernance"
	  //other-imports
)

Fix

```go

import (
	  "context"
	  abstractions "github.com/microsoft/kiota-abstractions-go"
	  msgraphsdk "github.com/microsoftgraph/msgraph-beta-sdk-go"
	  graphidentitygovernance "github.com/microsoftgraph/msgraph-beta-sdk-go/identitygovernance"
	  //other-imports
)
```

## Testing Instructions

Added unit tests

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 1]
```diff
@@ -92,17 +96,29 @@ private static string evaluateConfigPath(SnippetCodeGraph codeGraph)
...
+            return codeGraph.GetReferencedNamespaces(true).Select(x => evaluateNameSpaceName(x));
+        }
+
+
         private static string evaluateModelPath(SnippetCodeGraph codeGraph)
         {
             if (codeGraph.HasJsonBody())
             {
-                var path = codeGraph.Body.NamespaceName.Replace("microsoft.graph", "").Replace(".", "/");
+                var path = codeGraph.Body.NamespaceName.Replace("microsoft.graph", "").Replace(".", "/").Replace("//","/").ToLower().Split("/").FirstOrDefault();
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:110]: To lower invariant

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 2]
```diff
@@ -92,17 +96,29 @@ private static string evaluateConfigPath(SnippetCodeGraph codeGraph)
         }
 
 
+        private static IEnumerable<String> getModelsPaths(SnippetCodeGraph codeGraph)
+        {
+            // check the body and its children recursively for the namespaces
+            return codeGraph.GetReferencedNamespaces(true).Select(x => evaluateNameSpaceName(x));
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:102]:
```suggestion
            return codeGraph.GetReferencedNamespaces(true).Select(evaluateNameSpaceName);
```

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 3]
```diff
@@ -92,17 +96,29 @@ private static string evaluateConfigPath(SnippetCodeGraph codeGraph)
...
+                var path = codeGraph.Body.NamespaceName.Replace("microsoft.graph", "").Replace(".", "/").Replace("//","/").ToLower().Split("/").FirstOrDefault();
                 return path.EndsWith("/") ? path.Remove(path.Length - 1, 1) : path;
             }
 
             return string.Empty;
         }
 
+        private static string evaluateNameSpaceName(string nameSpace)
+        {
+            return nameSpace.Split(".").FirstOrDefault().ToLower().Replace(".microsoft.graph", "");
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:119]: To lower invariant

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs [hunk 4]
```diff
@@ -213,6 +214,41 @@ private static List<CodeProperty> parsePathParameters(SnippetModel snippetModel)
...
 
+
+        /// <summary>
+        /// Returns a list of all the namespaces that are referenced in the body.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the source enumerable.</typeparam>
+        public HashSet<String> GetReferencedNamespaces(Boolean shimmedMePackage = false)
+        {
+
+            HashSet<String> result = new HashSet<string>();
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:225]: Does casing matter here?

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs [hunk 5]
```diff
@@ -213,6 +214,41 @@ private static List<CodeProperty> parsePathParameters(SnippetModel snippetModel)
...
+            if (HasBody())
+            {
+                traverseProperty(Body, x =>
+                {
+                    if (!string.IsNullOrWhiteSpace(x.NamespaceName))
+                    {
+                        var nameSpaceName = shimmedMePackage ? x.NamespaceName.Split(".").Select(x => x.Equals("Me", StringComparison.OrdinalIgnoreCase) ? "Users" : x).Aggregate((current, next) => current + "." + next) : x.NamespaceName;
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:232]: String split option remove empty

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs [hunk 6]
```diff
@@ -229,7 +229,7 @@ public HashSet<String> GetReferencedNamespaces(Boolean shimmedMePackage = false)
                 {
                     if (!string.IsNullOrWhiteSpace(x.NamespaceName))
                     {
-                        var nameSpaceName = shimmedMePackage ? x.NamespaceName.Split(".").Select(x => x.Equals("Me", StringComparison.OrdinalIgnoreCase) ? "Users" : x).Aggregate((current, next) => current + "." + next) : x.NamespaceName;
+                        var nameSpaceName = shimmedMePackage ? x.NamespaceName.Split(".").Where(x => !string.IsNullOrEmpty(x)).Select(x => x.Equals("Me", StringComparison.OrdinalIgnoreCase) ? "Users" : x).Aggregate((current, next) => current + "." + next) : x.NamespaceName;
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:231]: this is what I meant here https://learn.microsoft.com/en-us/dotnet/api/system.string.split?view=net-8.0#system-string-split(system-char-system-stringsplitoptions)

**Review by baywet — DISMISSED** [review]
approving to avoid blocking this PR while I'm gone but I haven't had time to review the latest changes. Please have somebody else review it

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs (hunk 4 above)
Comment (andrueastman) [reply CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs]: If it doesn't, you can prevent potential duplicate entries due to different casing

```cs
HashSet<String> result = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

**Review by andrueastman — APPROVED** [review]

**Pull request merged**